Marlin/Marlin_main.cpp

```diff
@@ -2285,7 +2285,11 @@ static void clean_up_after_endstop_or_probe_move() {
         SERIAL_ECHOLNPAIR(" Discrepancy:", first_probe_z - current_position[Z_AXIS]);
       }
     #endif
-    return RAW_CURRENT_POSITION(Z) + zprobe_zoffset;
+    return RAW_CURRENT_POSITION(Z) + zprobe_zoffset
+    #if ENABLED(DELTA) // check if this is also required on cartesians
+      + home_offset[Z_AXIS]
```

Comment:
`RAW_CURRENT_POSITION(Z)` is `current_position[Z_AXIS] - home_offset[Z_AXIS] - position_shift[Z_AXIS]`.
So this results in `return current_position[Z_AXIS] - position_shift[Z_AXIS]`. It may be appropriate for a `DELTA`, which uses `home_offset[Z_AXIS]` as an offset to `DELTA_HEIGHT`, but it would not be appropriate for a cartesian, because we do no want any modifier added to the coordinate. We want a raw Z coordinate that has had all `workspace_offset` adjustment removed from it, so it is relative to `Z_HOME_POS`.